pkg/apis/multiclusterservice/v1/types.go

```diff
@@ -0,0 +1,62 @@
+package v1
+
+import (
+	"net"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//MultiClusterService describes the information of service that is available across
+//clusters. A shared informer running in the lighthouse control plane will listen for
+//service creation in every cluster that is part of the control plane and whenever a
+//service is created a MultiClusterService will be created. Lighthouse will
+//distribute the MultiClusterService to every other cluster that is the part of the
+//control plane. When the service is updated/deleted lighthouse ensures that the
+//updated state is reflected in every cluster.
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MultiClusterServiceSpec `json:"spec"`
+}
+
+//MultiClusterServiceSpec provides the specification of a MultiClusterService
+type MultiClusterServiceSpec struct {
+	//The name of the service. When a DNS request is received this field is used to
+	//retrieve the information about the clusters in which the service is present.
+	//This field is not optional.
+	ServiceName string `json:"service_name"`
+
+	//List of service information across clusters. This field is not optional. Each
+	//service name should have at least one service info associated with it.
+	Items []MultiClusterServiceInfo `json:"items"`
+}
+
+//MultiClusterServiceInfo provides information about service in each cluster.
+type MultiClusterServiceInfo struct {
+	//The unique identity of the cluster where the service is available. This field is
+	//not optional.
+	ClusterID string `json:"cluster_id"`
```

Comment:
It seems the convention in k8s is to use camel case for JSON field names and not use underscores, ie `cluster_id` -> `clusterID` etc.  (Sorry for not commenting on this earlier - I just noticed it)

Reply:
Done